Audio-Ingest-Outgest/MainComponent.cpp

```diff
@@ -0,0 +1,166 @@
+#include "MainComponent.h"
+
+MainComponent::MainComponent() :
+    juce::AudioAppComponent(otherDeviceManager),
+    state(Stopped), 
+    thumbnailCache(5),                           
+    thumbnail(512, formatManager, thumbnailCache),
+    openButton("Open"), playButton("Play"), stopButton("Stop") {
+
+    otherDeviceManager.initialise(2, 2, nullptr, true);
+    audioSettings.reset(new AudioDeviceSelectorComponent(otherDeviceManager, 2, 2, 2, 2, true, true, true, true));
+    setAudioChannels(2, 2);
+
+    setSize (400, 800);
+
+    //Initializing our button functions in this file
+    openButton.onClick = [this] {openButtonClicked();};
+
+    playButton.onClick = [this] {playButtonClicked();};
+    playButton.setColour(TextButton::buttonColourId, Colours::cornflowerblue);
+    playButton.setEnabled(true);
+
+    stopButton.onClick = [this] {stopButtonClicked();};
+    stopButton.setColour(TextButton::buttonColourId, Colours::orange);
+    stopButton.setEnabled(false);
+
+    addAndMakeVisible(audioSettings.get());
+    addAndMakeVisible(&visualiser);
+    addAndMakeVisible(&stopButton);
+    addAndMakeVisible(&playButton);
+    addAndMakeVisible(&openButton);
+
+    formatManager.registerBasicFormats();
+    if (RuntimePermissions::isRequired (RuntimePermissions::recordAudio)
+        && ! RuntimePermissions::isGranted (RuntimePermissions::recordAudio)) {
+        RuntimePermissions::request (RuntimePermissions::recordAudio,
+                                     [&] (bool granted) { if (granted)  setAudioChannels (2, 2); });
+    }
+    else {
+        setAudioChannels (2, 2);
+    }
+}
+
+MainComponent::~MainComponent() {
+    // This shuts down the audio device and clears the current audio source
+    shutdownAudio();
+}
+
+void MainComponent::prepareToPlay (int samplesPerBlockExpected, double sampleRate) {
+    //All the necessary clearing of our visual and data buffers before playback
+    visualiser.clear();
+    transport.prepareToPlay(samplesPerBlockExpected, sampleRate);
+}
+
+void MainComponent::openButtonClicked() {
+
+    DBG("open clicked");
+
+    //choosing file
+
+    FileChooser fileChooser ("Select a mp3, .wav, or .aiff file", File::getSpecialLocation(File::userHomeDirectory), "*.wav; *.aiff; *.mp3");
```

Comment:
Is there a way to not have all of these JUCE functions/objects imported automatically? As in, it would be much easier to manage if you had to do `juce::FileChooser` instead, so there would be a clear separation of what is from JUCE and what is your own.

I took a look and couldn't find anything easy. *shakes fist at c++ libraries*

Reply:
Also, this line is too long. 
```suggestion
    FileChooser fileChooser(
        "Select a mp3, .wav, or .aiff file",
        File::getSpecialLocation(File::userHomeDirectory),
        "*.wav; *.aiff; *.mp3"
        );
```